NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt

```diff
@@ -225,6 +226,78 @@ open class NeuroIDClassUnitTests {
     //    setNIDJobServiceManager - Used for mocking
 
     //   setTestURL
+
+    private fun setupAttemptedLoginTestEnvironment() {
+        // fake out the clock
+        mockkStatic(Calendar::class)
+        every {Calendar.getInstance().timeInMillis} returns 1
+        // make the logger not throw
+        val logger = mockk<NIDLogWrapper>()
+        every { logger.e(any(), any()) } just runs
+        NeuroID.getInternalInstance()?.logger = logger
+        // everything else as normal
+        setMockedDataStore()
+        setMockedNIDJobServiceManager(false)
+        NeuroID.isSDKStarted = true
+    }
+
+    private fun setupAttemptedLoginTestEnvironmentException() {
+        // fake out the clock
+        mockkStatic(Calendar::class)
+        every {Calendar.getInstance().timeInMillis} returns 1
+        // make the logger not throw
+        val logger = mockk<NIDLogWrapper>()
+        every { logger.e(any(), any()) } just runs
+        NeuroID.getInternalInstance()?.logger = logger
+        // make the datamanager throw exception
+        val dataStoreManager = mockk<NIDDataStoreManager>()
+        every{dataStoreManager.isFullBuffer()} returns false
+        every{dataStoreManager.saveEvent(any())} throws (Exception("save event exception"))
+        NeuroID.getInternalInstance()?.setDataStoreInstance(dataStoreManager)
+        // everything else as normal
+        setMockedNIDJobServiceManager(false)
+        NeuroID.isSDKStarted = true
+    }
+
+    private fun testAttemptedLogin(userId: String?, expectedHash: String, expectedResult: Boolean) {
+        setupAttemptedLoginTestEnvironment()
```

Comment:
nit-since uid is not being hashed it would be less confusing to change the var name `expectedHash`

Reply:
missed that. I can change it in the tests.